### PR TITLE
instance restart (reload) event

### PR DESCRIPTION
**Pull request by sirkon**
### 1. What does this change do, exactly?
New event type called *restart* added which is emitted at server restart (at SIGUSR1)

### 2. Please link to the relevant issues.
We are using caddy as embedded server and we need to make preparations before restarts

### 3. Which documentation changes (if any) need to be made because of this PR?
Nothing

### 4. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I am willing to help maintain this change if there are issues with it later

### CODE REVIEW

File: plugins.go [hunk 1]
```diff
@@ -263,9 +263,10 @@ type EventName string
 // Define names for the various events
 const (
 	StartupEvent         EventName = "startup"
-	ShutdownEvent        EventName = "shutdown"
-	CertRenewEvent       EventName = "certrenew"
-	InstanceStartupEvent EventName = "instancestartup"
+	ShutdownEvent                  = "shutdown"
+	CertRenewEvent                 = "certrenew"
+	InstanceStartupEvent           = "instancestartup"
+	InstanceRestart                = "restart"
```
Comment (mholt) [inline plugins.go]: Can we make the string consistent with the name? "instancerestart"?

Comment (sirkon) [reply plugins.go]: sure, done

File: plugins.go [hunk 2]
```diff
@@ -263,9 +263,10 @@ type EventName string
 // Define names for the various events
 const (
 	StartupEvent         EventName = "startup"
-	ShutdownEvent        EventName = "shutdown"
-	CertRenewEvent       EventName = "certrenew"
-	InstanceStartupEvent EventName = "instancestartup"
+	ShutdownEvent                  = "shutdown"
+	CertRenewEvent                 = "certrenew"
+	InstanceStartupEvent           = "instancestartup"
+	InstanceRestart                = "instancerestart"
```
Comment (lbguilherme) [inline plugins.go]: Maybe it also should be `InstanceRestartEvent` for consistency

Comment (sirkon) [reply plugins.go]: indeed, pushed

File: sigtrap_posix.go [hunk 3]
```diff
@@ -90,6 +90,7 @@ func trapSignalsPosix() {
 				purgeEventHooks()
 
 				// Kick off the restart; our work is done
+				EmitEvent(InstanceRestart, nil)
```
Comment (francislavoie) [inline sigtrap_posix.go]: You missed this one 😛 needs to be `InstanceRestartEvent` too

Comment (sirkon) [reply sigtrap_posix.go]: oops, fixed :)

**Comment by mholt** [comment]
Thanks!

**Pull request merged**